tkg/client/upgrade_cluster.go

```diff
@@ -1176,6 +1178,26 @@ func (c *TkgClient) patchKubernetesVersionToMachineDeployment(regionalClusterCli
 	return nil
 }
 
+func addAWSIngressrule(regionalClusterClient clusterclient.Client, upgradeClusterConfig *ClusterUpgradeInfo, newRule capav1beta2.CNIIngressRule) error {
+	awsClusterObject := &capav1beta2.AWSCluster{}
+	if err := regionalClusterClient.GetResource(awsClusterObject, upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace, nil, nil); err != nil {
+		return errors.Wrap(err, "unable to retrieve aws cluster object")
+	}
+	if awsClusterObject.Spec.NetworkSpec.SecurityGroupOverrides != nil {
```

Comment:
Codecov is a bit misleading here.  I believe what it means is that the "real"  GetResource function does not get called. 
However the tests in uprade_cluster_test.go do touch this lines. Old code should do it. And if not the new tests added do for sure. (see tkg/client/upgrade_cluster.go in this patch)